src/cascadia/TerminalSettingsModel/ApplicationState.cpp

```diff
@@ -309,6 +309,35 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         _throttler();
     }
 
+    bool ApplicationState::DismissBadge(const hstring& badgeId)
+    {
+        bool inserted{ false };
+        {
+            const auto state = _state.lock();
+            if (!state->DismissedBadges)
+            {
+                state->DismissedBadges = std::unordered_set<hstring>{};
+            }
+            if (!state->DismissedBadges->contains(badgeId))
+            {
+                state->DismissedBadges->insert(badgeId);

```

Comment:
Insert returns a tuple/pair whose second member is a bool. That bool is true if the item got inserted. This would simplify the code here because you don't need to call contains() first.